### PR TITLE
Police outfits & M6B

### PR DESCRIPTION
**Pull request by Desolane**
:cl: Desolane900
rscadd: Added new GCPD outfits.
rscadd: Added new GCPD helmets.
rscadd: Added new GCPD armors.
rscadd: Added new helmet subtype: gas.
rscadd: Added the M6B and police variant.
rscdel: Nukes old GCPD outfits.
/:cl:

### CODE REVIEW

**Comment by Desolane** [comment]
This is technically merge capable as is, but I am still waiting for magazine sprites for the M6B before I want it merged. Peer review on the gasmask helmet subtype would be appreciated as it functions and works incredibly well for everything from tear gas to pepperspray, it's probably not in the ideal place nor should it be a subtype the way it currently is, but it needed done and I dunno how else I would have done it.

File: code/modules/halo/clothing/civilian.dm [hunk 1]
```diff
@@ -1,19 +1,156 @@
...
+	valid_accessory_slots = list(ACCESSORY_SLOT_INSIGNIA)//the badge
 	body_parts_covered = UPPER_TORSO|LOWER_TORSO|ARMS
-	armor = list(melee = 40, bullet = 35, laser = 15, energy = 15, bomb = 40, bio = 0, rad = 0)
+	armor = list(melee = 40, bullet = 35, laser = 15, energy = 15, bomb = 40, bio = 20, rad = 15)//energy resist to stay nerfed, let the UNSC fight covies not the police. -Deso
+	icon = 'code/modules/halo/clothing/gcpd.dmi'
+	item_icons = list(
+		slot_l_hand_str = 'code/modules/halo/clothing/gcpd_leftinhands.dmi',
+		slot_r_hand_str = 'code/modules/halo/clothing/gcpd_rightinhands.dmi',
+		)
+	armor_thickness = 35//this had zero thickness lol
```
Comment (X0-11) [inline code/modules/halo/clothing/civilian.dm]: no it didn't, it had 20 as per inheriting from /armor.
35 is way too high. Adding more armor thickness than 20 has a number of effects including making the person have more grace shots. If you want to make the armor "heavy" armor, increase the base armor values which act as "minimums"

File: code/modules/halo/clothing/civilian.dm [hunk 2]
```diff
@@ -1,19 +1,156 @@
...
+
+/obj/item/clothing/suit/armor/vest/police/New()
+	..()
+	slowdown_per_slot[slot_wear_suit] = 2
+
+/obj/item/clothing/suit/armor/vest/police_medium//medium armor
+	name = "medium ballistic vest"
+	desc = "A heavy vest designed for dangerous operations that features full upper and lower torso coverage. For room clearing and armed civilians."
+	armor = list(melee = 35, bullet = 30, laser = 15, energy = 15, bomb = 30, bio = 10, rad = 15)
+	armor_thickness = 15
```
Comment (X0-11) [inline code/modules/halo/clothing/civilian.dm]: this should probably be the base 20 value

File: code/modules/halo/clothing/civilian.dm [hunk 3]
```diff
@@ -1,19 +1,156 @@
...
+/obj/item/clothing/head/helmet/swat/police/medium//<
+	icon_state = "gcpd-helmet-googles"
+	item_state = "gcpd-helmet-googles"
+	name = "ballistic helmet"
+	desc = "Heavier helmet designed to be worn with the medium ballistic vest with more focus on protecting the eyes and head from shrapnel and bullets."
+	armor = list(melee = 20, bullet = 30, laser = 10, energy = 10, bomb = 30, bio = 10, rad = 15)
+	item_flags = THICKMATERIAL
+	body_parts_covered = HEAD|EYES
+
+/obj/item/clothing/head/helmet/gas//copied code from gas masks and applied it to helmets for variability //not sure if best place for it
```
Comment (X0-11) [inline code/modules/halo/clothing/civilian.dm:117]: gasmaskhelm might not work because it's likely that the helmet doesn't get checked for filtering gases

Comment (Desolane) [reply code/modules/halo/clothing/civilian.dm:117]: The proc checks for gas differences, it's just copypasta from actual gas masks.

Comment (X0-11) [reply code/modules/halo/clothing/civilian.dm:117]: yes but does the helmet ever get the filter_gas check actually called on it when processing breaths

Comment (Desolane) [reply code/modules/halo/clothing/civilian.dm:117]: Line 122

Comment (X0-11) [reply code/modules/halo/clothing/civilian.dm:117]: yes the proc exists but does breathcode care or actually call it anywhere. it's likely that breathcode only calls such a proc on the maskslot

File: code/modules/halo/clothing/civilian.dm [hunk 4]
```diff
@@ -1,19 +1,163 @@
...
+
+/obj/item/clothing/suit/armor/vest/police_medium/New()//speeds are factoring other gear and chasing people on foot, meant to encourage using lighter armors to reduce meta
+	..()
+	slowdown_per_slot[slot_wear_suit] = -1
+
+/obj/item/clothing/suit/storage/vest/tactical/police//light armor
+	name = "light ballistic vest"
+	desc = "Lightweight ballistic vest designed to reduce damage from low caliber rounds or stab wounds specifically to the upper torso where your important guts are. For your average beatcop."
+	armor = list(melee = 35, bullet = 20, laser = 10, energy = 10, bomb = 20, bio = 10, rad = 15)
+	armor_thickness = 3
```
Comment (X0-11) [inline code/modules/halo/clothing/civilian.dm:45]: I feel like you've misunderstood armor_thickness again.
armor_thickness / 10 = multiplier to armor values
20 gives about 2 grace shots.

**Review by X0-11 — CHANGES_REQUESTED** [review]

**Review by BDpuffy420 — APPROVED** [review]

**Pull request merged**